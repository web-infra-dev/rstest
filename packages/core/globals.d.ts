declare global {
  const test: typeof import('@rstest/core')['test'];
  const describe: typeof import('@rstest/core')['describe'];
  const it: typeof import('@rstest/core')['it'];
  const expect: typeof import('@rstest/core')['expect'];
}
export {};
