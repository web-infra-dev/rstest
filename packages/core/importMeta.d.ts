interface ImportMeta {
  readonly rstest?: typeof import('@rstest/core');
}
