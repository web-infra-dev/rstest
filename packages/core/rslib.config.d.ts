import { RslibConfig } from '@rslib/core';
declare const config: RslibConfig;
export default config;
