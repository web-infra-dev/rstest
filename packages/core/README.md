<picture>
  <img alt="Rstest Banner" src="https://assets.rspack.dev/rstest/rstest-banner.png">
</picture>

# Rstest

Rstest is a testing framework powered by Rspack.

## ðŸ“– License

Rstest is licensed under the [MIT License](https://github.com/web-infra-dev/rstest/blob/main/LICENSE).
