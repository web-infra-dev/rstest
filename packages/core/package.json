{
  "name": "@rstest/core",
  "version": "0.0.0",
  "description": "The Rsbuild-based test tool.",
  "bugs": {
    "url": "https://github.com/web-infra-dev/rstest/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/web-infra-dev/rstest",
    "directory": "packages/core"
  },
  "license": "MIT",
  "type": "module",
  "main": "./dist/index.js",
  "types": "./dist-types/index.d.ts",
  "bin": {
    "rstest": "./bin/rstest.js"
  },
  "exports": {
    ".": {
      "types": "./dist-types/index.d.ts",
      "default": "./dist/index.js"
    },
    "./globals": {
      "types": "./globals.d.ts"
    }
  },
  "files": [
    "bin",
    "dist",
    "dist-types",
    "compiled",
    "types.d.ts"
  ],
  "scripts": {
    "build": "rslib build",
    "dev": "rslib build --watch",
    "test": "npx rstest run --globals"
  },
  "dependencies": {
    "@rsbuild/core": "^1.2.19",
    "birpc": "2.2.0",
    "chai": "^5.2.0",
    "std-env": "^3.8.1",
    "tinypool": "^1.0.2"
  },
  "devDependencies": {
    "@babel/code-frame": "^7.26.2",
    "@jridgewell/trace-mapping": "0.3.25",
    "@rslib/core": "0.5.5",
    "@rstest/tsconfig": "workspace:*",
    "@types/babel__code-frame": "^7.0.6",
    "@vitest/expect": "^3.0.9",
    "cac": "^6.7.14",
    "picocolors": "^1.1.1",
    "rslog": "^1.2.3",
    "stacktrace-parser": "0.1.11",
    "tinyglobby": "^0.2.12"
  },
  "engines": {
    "node": ">=16.7.0"
  },
  "publishConfig": {
    "access": "public",
    "provenance": true,
    "registry": "https://registry.npmjs.org/"
  }
}
