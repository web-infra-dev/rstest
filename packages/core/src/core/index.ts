import type { RstestCommand, RstestConfig, RstestInstance } from '../types';
import { createContext } from './context';

export function createRstest(
  config: RstestConfig,
  command: RstestCommand,
  fileFilters: string[],
): RstestInstance {
  const context = createContext({ cwd: process.cwd(), command }, config);

  const runTests = async (): Promise<void> => {
    const { runTests } = await import('./runTests');
    await runTests(context, fileFilters);
  };

  return {
    runTests,
  };
}
