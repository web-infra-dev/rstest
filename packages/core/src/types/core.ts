import type { SnapshotManager } from '@vitest/snapshot/manager';
import type { NormalizedConfig, RstestConfig } from './config';
import type { Reporter } from './reporter';

export type RstestCommand = 'watch' | 'run';

export type RstestContext = {
  /** The Rstest core version. */
  version: string;
  /** The root path of current project. */
  rootPath: string;
  /** The original Rstest config passed from the createRstest method. */
  originalConfig: Readonly<RstestConfig>;
  /** The normalized Rstest config. */
  normalizedConfig: NormalizedConfig;
  /**
   * The command type.
   *
   * - dev: `rstest dev`
   * - run: `rstest run`
   */
  command: RstestCommand;
  reporters: Reporter[];
  snapshotManager: SnapshotManager;
};

export type RstestInstance = {
  runTests: () => Promise<void>;
};
