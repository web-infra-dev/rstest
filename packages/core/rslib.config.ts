import { defineConfig } from '@rslib/core';

export default defineConfig({
  lib: [
    {
      format: 'esm',
      syntax: ['node 16'],
      dts: {
        bundle: false,
        distPath: './dist-types',
      },
      output: {
        minify: {
          jsOptions: {
            minimizerOptions: {
              mangle: false,
              minify: false,
              compress: {
                defaults: false,
                unused: true,
                dead_code: true,
                toplevel: true,
                // fix `Couldn't infer stack frame for inline snapshot` error
                // should keep function name used to filter stack trace
                keep_fnames: true,
              },
              format: {
                comments: 'some',
                preserve_annotations: true,
              },
            },
          },
        },
      },
      source: {
        entry: {
          public: './src/public.ts',
          node: './src/node.ts',
          cli: './src/cli/index.ts',
          worker: './src/runtime/worker/index.ts',
        },
        define: {
          RSTEST_VERSION: JSON.stringify(require('./package.json').version),
        },
      },
    },
  ],
  tools: {
    rspack: {
      watchOptions: {
        ignored: /\.git/,
      },
    },
  },
});
