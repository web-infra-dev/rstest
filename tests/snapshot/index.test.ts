import { describe, expect, it } from '@rstest/core';

describe('test snapshot', () => {
  it('test toMatchSnapshot API', () => {
    expect('hello world').toMatchSnapshot();
  });

  it.todo('test toMatchInlineSnapshot API', () => {
    expect('hello world').toMatchInlineSnapshot();
  });
});
