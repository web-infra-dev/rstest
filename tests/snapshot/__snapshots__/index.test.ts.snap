// Snapshot v1

exports[`test toMatchInlineSnapshot API 1`] = `"hello world"`;
